src/o365/spo/commands/listitem/listitem-record-declare.ts

```diff
@@ -0,0 +1,309 @@
+    }
+    else {
+      requestBody = `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><StaticMethod TypeId="{ea8e1356-5910-4e69-bc05-d0c30ed657fc}" Name="DeclareItemAsRecord" Id="37"><Parameters><Parameter ObjectPathId="12" /></Parameters></StaticMethod></Actions><ObjectPaths><Identity Id="12" Name="${webIdentity}:list:${listId}:item:${id},1" /></ObjectPaths></Request>`;
+    }
+
+    return requestBody;
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: "-u, --webUrl <webUrl>",
+        description: "URL of the site where the item should be added"
+      },
+      {
+        option: "-l, --listId [listId]",
+        description:
+          "ID of the list where the item should be added. Specify listId or listTitle but not both"
+      },
+      {
+        option: "-t, --listTitle [listTitle]",
+        description:
+          "Title of the list where the item should be added. Specify listId or listTitle but not both"
+      },
+      {
+        option: "-i, --id [id]",
+        description: "ID of the list item to declare as a record"
+      },
+      {
+        option: "-d, --date [date]",
+        description: "Record declaration date"
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public types(): CommandTypes {
+    return {
+      string: ["webUrl", "listId", "listTitle", "id", "date"]
+    };
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (!args.options.webUrl) {
+        return "Required parameter webUrl missing";
+      }
+
+      const isValidSharePointUrl:
+        | boolean
+        | string = SpoCommand.isValidSharePointUrl(args.options.webUrl);
+      if (isValidSharePointUrl !== true) {
+        return isValidSharePointUrl;
+      }
+
+      if (!args.options.listId && !args.options.listTitle) {
+        return `Specify listId or listTitle`;
+      }
+
+      if (args.options.listId && args.options.listTitle) {
+        return `Specify listId or listTitle but not both. (${args.options.listId} | ${args.options.listTitle})`;
+      }
+
+      if (args.options.listId && !Utils.isValidGuid(args.options.listId)) {
+        return `${args.options.listId} in option listId is not a valid GUID`;
+      }
+
+      if (!args.options.id) {
+        return `Specify id for item to declare as a record`;
+      }
+
+      if (args.options.id && !Utils.isPositiveInt(args.options.id)) {
+        return `Specify id for item to declare as a record`;
```

Comment:
This error message should clearly indicate that the id has been specified but is not a valid number

Reply:
Updated